### ISSUE TITLE
[Java] swagger-codegen v3.0.9 throws NPE on API spec processed successfully by v3.0.8

### ISSUE THREAD

**Issue by chrisinmtown** [swagger-api/swagger-codegen#9541]
##### Description

I have a somewhat complex spec (see below) that can be processed successfully by the following configuration using version 3.0.8, but not by 3.0.9. Here's a partial stack trace from the error:

```
java.lang.RuntimeException: Could not process model 'Response_xnb_choiceNrMode_fdd_dlarFcnFreqInfo_sulInformation_sulTransmissionBandwidth'.Please make sure that your schema is correct!
    at io.swagger.codegen.v3.DefaultGenerator.generateModels (DefaultGenerator.java:378)

....

Caused by: java.lang.NullPointerException
    at io.swagger.codegen.v3.generators.util.OpenAPIUtil.getSimpleRef (OpenAPIUtil.java:32)
    at io.swagger.codegen.v3.generators.SchemaHandler.createFromOneOfSchemas (SchemaHandler.java:131)
    at io.swagger.codegen.v3.generators.SchemaHandler.createCodegenModel (SchemaHandler.java:30)
...
```
I use a maven-based build with the following configuration. All the ${foo} things are just strings, anything will do.

```
                        <plugin>
                                <!-- This 2019 version is required for OpenAPI 3 -->
                                <groupId>io.swagger.codegen.v3</groupId>
                                <artifactId>swagger-codegen-maven-plugin</artifactId>
                                <version>3.0.8</version>
                                <executions>
                                        <execution>
                                                <goals>
                                                        <goal>generate</goal>
                                                </goals>
                                                <configuration>
                                                        <inputSpec>${project.basedir}/src/main/resources/thespec.yaml</inputSpec>
                                                        <language>java</language>
                                                        <packageName>${client.base.package.name}</packageName>
                                                        <modelPackage>${client.base.package.name}.model</modelPackage>
                                                        <apiPackage>${client.base.package.name}.api</apiPackage>
                                                        <invokerPackage>${client.base.package.name}.invoker</invokerPackage>
                                                        <configOptions>
                                                                <groupId>${project.groupId}</groupId>
                                                                <artifactId>${project.artifactId}</artifactId>
                                                                <artifactVersion>${project.version}</artifactVersion>
                                                                <library>resttemplate</library>
                                                                <java8>true</java8>
                                                                <dateLibrary>java8</dateLibrary>
                                                                <licenseName>Apache 2.0</licenseName>
                                                                <licenseUrl>https://www.apache.org/licenses/LICENSE-2.0</licenseUrl>
                                                        </configOptions>
                                                </configuration>
                                        </execution>
                                </executions>
                        </plugin>

```

##### Swagger-codegen version

Using 3.0.9, so yes this appears to be a regression backwards in functionality.

##### Command line used for generation

mvn clean install

##### Steps to reproduce

Attempt to generate via mvn clean install

##### Related issues/PRs

I found no related issues.

##### Swagger declaration file content or url

Sorry for the length, I cut it down as far as I could, a single endpoint and a single schema element.

```yaml
openapi: 3.0.0
info:
  title: Service
  version: 1.0.0
servers:
  - url: 'http://{apiRoot}/v1'
    variables:
      apiRoot:
        default: 'localhost:3800'
paths:
  '/nodeb/{ranName}':
    get:
      tags:
        - nodeb
      summary: Get node by name
      operationId: getNb
      parameters:
        - name: ranName
          in: path
          required: true
          description: Name of RAN to return
          schema:
            type: string
      responses:
        '200':
          description: Successful operation
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Response'
components:
  schemas:
    Response:
      properties:
        xnb:
          properties:
            servedNrCells:
              items:
                properties:
                  nrNeighbourInfos:
                    items:
                      properties:
                        choiceNrMode:
                          properties:
                            fdd:
                              properties:
                                dlarFcnFreqInfo:
                                  properties:
                                    frequencyBands:
                                      items:
                                        properties:
                                          nrFrequencyBand:
                                            type: integer
                                          supportedSulBands:
                                            items:
                                              type: integer
                                            type: array
                                        additionalProperties: false
                                        type: object
                                      type: array
                                    nrArFcn:
                                      oneOf:
                                        - type: integer
                                    sulInformation:
                                      properties:
                                        sulArFcn:
                                          oneOf:
                                            - type: integer
                                        sulTransmissionBandwidth:
                                          properties:
                                            ncnrb:
                                              oneOf:
                                                - type: string
                                                - type: integer
                                            nrscs:
                                              oneOf:
                                                - type: string
                                                - type: integer
                                          additionalProperties: false
                                          type: object
                                      additionalProperties: false
                                      type: object
                                  additionalProperties: false
                                  type: object
                                ularFcnFreqInfo:
                                  properties:
                                    frequencyBands:
                                      items:
                                        properties:
                                          nrFrequencyBand:
                                            type: integer
                                          supportedSulBands:
                                            items:
                                              type: integer
                                            type: array
                                        additionalProperties: false
                                        type: object
                                      type: array
                                    nrArFcn:
                                      oneOf:
                                        - type: integer
                                    sulInformation:
                                      properties:
                                        sulArFcn:
                                          oneOf:
                                            - type: integer
                                        sulTransmissionBandwidth:
                                          properties:
                                            ncnrb:
                                              oneOf:
                                                - type: string
                                                - type: integer
                                            nrscs:
                                              oneOf:
                                                - type: string
                                                - type: integer
                                          additionalProperties: false
                                          type: object
                                      additionalProperties: false
                                      type: object
                                  additionalProperties: false
                                  type: object
                              additionalProperties: false
                              type: object
                            tdd:
                              properties:
                                arFcnNrFreqInfo:
                                  properties:
                                    frequencyBands:
                                      items:
                                        properties:
                                          nrFrequencyBand:
                                            type: integer
                                          supportedSulBands:
                                            items:
                                              type: integer
                                            type: array
                                        additionalProperties: false
                                        type: object
                                      type: array
                                    nrArFcn:
                                      oneOf:
                                        - type: integer
                                    sulInformation:
                                      properties:
                                        sulArFcn:
                                          oneOf:
                                            - type: integer
                                        sulTransmissionBandwidth:
                                          properties:
                                            ncnrb:
                                              oneOf:
                                                - type: string
                                                - type: integer
                                            nrscs:
                                              oneOf:
                                                - type: string
                                                - type: integer
                                          additionalProperties: false
                                          type: object
                                      additionalProperties: false
                                      type: object
                                  additionalProperties: false
                                  type: object
                              additionalProperties: false
                              type: object
                          additionalProperties: false
                          type: object
                        configuredStac:
                          type: string
                        nrCgi:
                          type: string
                        nrMode:
                          oneOf:
                            - type: string
                            - type: integer
                        nrPci:
                          type: integer
                        stac5g:
                          type: string
                      additionalProperties: false
                      type: object
                    type: array
                  servedNrCellInformation:
                    properties:
                      cellId:
                        type: string
                      choiceNrMode:
                        properties:
                          fdd:
                            properties:
                              dlFreqInfo:
                                properties:
                                  frequencyBands:
                                    items:
                                      properties:
                                        nrFrequencyBand:
                                          type: integer
                                        supportedSulBands:
                                          items:
                                            type: integer
                                          type: array
                                      additionalProperties: false
                                      type: object
                                    type: array
                                  nrArFcn:
                                    oneOf:
                                      - type: integer
                                  sulInformation:
                                    properties:
                                      sulArFcn:
                                        oneOf:
                                          - type: integer
                                      sulTransmissionBandwidth:
                                        properties:
                                          ncnrb:
                                            oneOf:
                                              - type: string
                                              - type: integer
                                          nrscs:
                                            oneOf:
                                              - type: string
                                              - type: integer
                                        additionalProperties: false
                                        type: object
                                    additionalProperties: false
                                    type: object
                                additionalProperties: false
                                type: object
                              dlTransmissionBandwidth:
                                properties:
                                  ncnrb:
                                    oneOf:
                                      - type: string
                                      - type: integer
                                  nrscs:
                                    oneOf:
                                      - type: string
                                      - type: integer
                                additionalProperties: false
                                type: object
                              ulFreqInfo:
                                properties:
                                  frequencyBands:
                                    items:
                                      properties:
                                        nrFrequencyBand:
                                          type: integer
                                        supportedSulBands:
                                          items:
                                            type: integer
                                          type: array
                                      additionalProperties: false
                                      type: object
                                    type: array
                                  nrArFcn:
                                    oneOf:
                                      - type: integer
                                  sulInformation:
                                    properties:
                                      sulArFcn:
                                        oneOf:
                                          - type: integer
                                      sulTransmissionBandwidth:
                                        properties:
                                          ncnrb:
                                            oneOf:
                                              - type: string
                                              - type: integer
                                          nrscs:
                                            oneOf:
                                              - type: string
                                              - type: integer
                                        additionalProperties: false
                                        type: object
                                    additionalProperties: false
                                    type: object
                                additionalProperties: false
                                type: object
                              ulTransmissionBandwidth:
                                properties:
                                  ncnrb:
                                    oneOf:
                                      - type: string
                                      - type: integer
                                  nrscs:
                                    oneOf:
                                      - type: string
                                      - type: integer
                                additionalProperties: false
                                type: object
                            additionalProperties: false
                            type: object
                          tdd:
                            properties:
                              nrFreqInfo:
                                properties:
                                  frequencyBands:
                                    items:
                                      properties:
                                        nrFrequencyBand:
                                          type: integer
                                        supportedSulBands:
                                          items:
                                            type: integer
                                          type: array
                                      additionalProperties: false
                                      type: object
                                    type: array
                                  nrArFcn:
                                    oneOf:
                                      - type: integer
                                  sulInformation:
                                    properties:
                                      sulArFcn:
                                        oneOf:
                                          - type: integer
                                      sulTransmissionBandwidth:
                                        properties:
                                          ncnrb:
                                            oneOf:
                                              - type: string
                                              - type: integer
                                          nrscs:
                                            oneOf:
                                              - type: string
                                              - type: integer
                                        additionalProperties: false
                                        type: object
                                    additionalProperties: false
                                    type: object
                                additionalProperties: false
                                type: object
                              transmissionBandwidth:
                                properties:
                                  ncnrb:
                                    oneOf:
                                      - type: string
                                      - type: integer
                                  nrscs:
                                    oneOf:
                                      - type: string
                                      - type: integer
                                additionalProperties: false
                                type: object
                            additionalProperties: false
                            type: object
                        additionalProperties: false
                        type: object
                      configuredStac:
                        type: string
                      nrMode:
                        oneOf:
                          - type: string
                          - type: integer
                      nrPci:
                        type: integer
                      servedPlmns:
                        items:
                          type: string
                        type: array
                      stac5g:
                        type: string
                    additionalProperties: false
                    type: object
                additionalProperties: false
                type: object
              type: array
          additionalProperties: false
          type: object

```

**Comment by chris-prop** [comment]
I think this is due to the updated `oneOf`, `anyOf`, `allOf` processing. It doesn't handle native types, only references. I've fixed it by updating updating `SchenaHandler` with
```java
    private CodegenModel createFromOneOfSchemas(List<Schema> schemas) {
        final CodegenModel codegenModel = CodegenModelFactory.newInstance(CodegenModelType.MODEL);
        final List<String> modelNames = new ArrayList<>();

        for (Schema interfaceSchema : schemas) {
            if (StringUtils.isNotBlank(interfaceSchema.get$ref())) {
                String schemaName = OpenAPIUtil.getSimpleRef(interfaceSchema.get$ref());
                modelNames.add(codegenConfig.toModelName(schemaName));
            }
        }
        codegenModel.vendorExtensions.put("x-model-names", modelNames);
        return codegenModel;
    }
```

**Comment by frantuma** [comment]
Thanks for reporting and analyzing this! It has been fixed in #415, closing ticket, please reopen if you're still experiencing issues

**Comment by chrisinmtown** [comment]
Thanks for the quick fix. Will this be released as 3.0.10?
